base_event.cpp

```diff
@@ -1,92 +1,153 @@
-      
-  //std::string filepath = (filepathText->GetValue()).ToStdString();
-//  printToOutput(filepath);
-  
-  // so far only xyz files allowed
-  Ctrl::getInstance()->runCalculation();
-  return;
-}
-
-// browse for atom file
-void MainFrame::OnAtomBrowse(wxCommandEvent& event){
-  std::string filetype = "XYZ files (*.xyz)|*.xyz";
-  OnBrowse(filetype, filepathText);
-  return;
-}
-
-// browse for radius file
-void MainFrame::OnRadiusBrowse(wxCommandEvent& event){
-  std::string filetype = "TXT files (*.txt)|*.txt";
-  OnBrowse(filetype, radiuspathText);
-  return;
-}
-
-// browse (can only be called by another method function)
-void MainFrame::OnBrowse(std::string& filetype, wxTextCtrl* textbox){
-  wxFileDialog openFileDialog
-    (this, 
-     _("Select File"), 
-     "", 
-     "",
-     filetype, 
-     wxFD_OPEN|wxFD_FILE_MUST_EXIST,
-     wxDefaultPosition,
-     wxDefaultSize,
-     "file browser"
-    );
-   
-  // if user closes dialogue
-  if (openFileDialog.ShowModal() == wxID_CANCEL)
-    return;
-    
-  // proceed loading the file chosen by the user;
-  wxFileInputStream input_stream(openFileDialog.GetPath());
-  // if filepath is invalid
-  if (!input_stream.IsOk())
-  {
-    wxLogError("Cannot open file '%s'.", openFileDialog.GetPath());
-    return;
-  }
-  textbox->SetLabel(openFileDialog.GetPath());
-  return;
-}
+#include <wx/wxprec.h>
+
+#ifndef WX_PRECOMP
+#  include <wx/wx.h>
+#endif
+
+#include "base.h"
+#include "controller.h"
+#include <vector>
+#include <utility>
+
+/////////////////
+// EVENT TABLE //
+/////////////////
+
+BEGIN_EVENT_TABLE(MainFrame, wxFrame)
+  EVT_BUTTON(BUTTON_Calc, MainFrame::OnCalc)
+  EVT_BUTTON(BUTTON_Browse, MainFrame::OnAtomBrowse)
+  EVT_BUTTON(BUTTON_Radius, MainFrame::OnRadiusBrowse)
+  EVT_BUTTON(BUTTON_LoadFiles, MainFrame::OnLoadFiles)
+  EVT_GRID_CELL_CHANGING(MainFrame::GridChange)
+END_EVENT_TABLE()
+
+////////////////////////////////
+// METHODS FOR EVENT HANDLING //
+////////////////////////////////
+
+// exit program
+void MainFrame::OnExit(wxCommandEvent& event)
+{
+  this->Close(TRUE);
+}
+
+// display text in output, used for debugging
+void MainFrame::OnPrint(wxCommandEvent& event)
+{
+  std::string text = "treat yourself well";
+  printToOutput(text);
+  return;
+}
+
+// begin calculation
+void MainFrame::OnCalc(wxCommandEvent& event){
+
+  //std::string filepath = (filepathText->GetValue()).ToStdString();
+  //  printToOutput(filepath);
+
+  // Deactivate the Start and Load files buttons during the calculation
+  MainFrame::calcButton->Enable(false);
+  MainFrame::loadFilesButton->Enable(false);
+  // Prevents editing of the atom list grid during the calculation
+  MainFrame::atomListGrid->EnableEditing(false);
+  // so far only xyz files allowed
+  Ctrl::getInstance()->runCalculation();
+  // without wxYield, the button is grayed but still records clicks
+  // yet, wxYield is apparently dangerous in an event handler, need to find an alternative
+  wxYield();

```

Comment:
I'll call the functions that control the GUI elements in the controller class. Since I don't know what this does, I'll comment it out for now, and you'll likely have to reinclude it. Although your comments make me thing that it really shouldn't be here.

Reply:
After googling this issue about disabled buttons still recording clicks and queuing them until the button is enabled again, this is the simplest solution that came up (wxYield).
It did the job and didn't cause any crash or decrease and efficiency so I kept it for now.
But I have to admit that we might want to find a different solution later on.